api/v1alpha1/nodepool_types.go

```diff
@@ -286,6 +287,19 @@ type NodePoolPlatform struct {
 
 	// IBMCloud defines IBMCloud specific settings for components
 	IBMCloud *IBMCloudPlatformSpec `json:"ibmcloud,omitempty"`
+
+	// Kubevirt specifies the configuration used when operating on KubeVirt platform.
+	//
+	// +optional
+	// +immutable
+	Kubevirt *KubevirtNodePoolPlatform `json:"kubevirt,omitempty"`
+}
+
+// KubevirtNodePoolPlatform specifies the configuration of a NodePool when operating
+// on KubeVirt platform.
+type KubevirtNodePoolPlatform struct {
+	// NodeTemplate Spec contains the VirtualMachineInstance specification.
+	NodeTemplate *capikubevirt.VirtualMachineTemplateSpec `json:"nodetemplate,omitempty"`
```

Comment:
I think to be consistent the json filed should be named `nodeTemplate` (capital T).

Reply:
Fixed